### DISCUSSION TITLE
Looking for cause of black screen/Decompile scripts fix

### DISCUSSION THREAD

**Discussion by Michael21107** [scripthookvdotnet/scripthookvdotnet#1362] [Q&A] [answered]
i have a few scripts including a few decompiled ones (.cs) too and when i launched the game i had a black screen, i could move and hear but ''see'' only with pause menu opened, could someone perhaps find the cause of that?
![image](https://github.com/scripthookvdotnet/scripthookvdotnet/assets/121039033/046cdf05-3dfb-43b8-ae45-6bdf01a35d49)
[ScriptHookVDotNet.log](https://github.com/scripthookvdotnet/scripthookvdotnet/files/13666453/ScriptHookVDotNet.log)

**Comment by kagikn** [comment, marked as answer]
I think LastLocation.dll was the cause? At least log says SHVDN stops running to because it took too long time in a single tick, probably because the author didn't care about how it runs too much about where it stopped in its code.

ForestShelter.cs and InteriorsAdditions.cs failed to compile btw. If you see "Failed to compile" for scripts you could run without complication issues in v3.6.0, add ".2" the scripts the lines mention before the extension names (probably ".cs"), like `SomeScript.2.cs`. That will tell SHVDN to compile them against v2 API like before.

**Reply by Michael21107** [reply]
yep looks like it, thanks and i will try that .2

**Reply by Michael21107** [reply]
should i rename the 2 cs scripts ini's too?

**Reply by kagikn** [reply]
Ah didn't think about ini, but it depends on how cs scripts are written (you can view the contents with a text editor). If scripts don't work because you have renamed the ini files as well, you'll have to revert the name changes of ini files. Consult a script dev who knows how to write SHVDN scripts may help you understand what you should do about file names of ini files though (by checking how scripts are written).

**Reply by Michael21107** [reply]
i guess this means that i shouldn't rename the .ini, right?
![image](https://github.com/scripthookvdotnet/scripthookvdotnet/assets/121039033/23045518-57f4-47d0-a467-9fbe6bf092b9)

**Reply by kagikn** [reply]
yes, in this case you shouldn't

**Reply by Michael21107** [reply]
gotcha, thanks, perhaps u can also find the source of this error? happened in like a minute from all other stuff 
[ScriptHookVDotNet.log](https://github.com/scripthookvdotnet/scripthookvdotnet/files/13666777/ScriptHookVDotNet.log)
`[23:51:08] [ERROR] Caught unhandled exception:
System.NullReferenceException: Object reference not set to an instance of an object.
   at Cycles.OnTick(Object sender, EventArgs e)
   at SHVDN.Script.DoTick()
[23:51:08] [ERROR] The exception was thrown while executing the script Cycles.
[23:51:08] [WARNING] Aborted script Cycles.
   at System.Threading.Thread.AbortInternal()
   at System.Threading.Thread.Abort()
   at SHVDN.Script.Abort()
   at SHVDN.Script.DoTick()
   at SHVDN.Script.MainLoop()`

**Reply by kagikn** [reply]
Nah, I can't find the cause how `Cycles` class throws an `NullReferenceException`, without inspecting what the internal IL code is. I guess `Cycles` is a part of `MP Vehicles in SP.dll` since the dll has a lot of script classes unlike the other dlls.

Looks like we should make SHVDN log the name of the file where the crashed script is in, so you can quickly identify problematic scripts report crashing issues to the authors.

**Reply by Michael21107** [reply]
thank you and yes, that's a great idea (meaning you and other contributors should do it, cuz if i did it, it would crash gta online for everyone too lol)
